### DISCUSSION TITLE
Where to put the env vars file?

### DISCUSSION THREAD

**Discussion by droplinxuser** [shlinkio/shlink#1971] [Help wanted] [answered]
### Shlink version

3.7.3

### PHP version

8.3.1

### How do you serve Shlink

Self-hosted RoadRunner

### Database engine

MariaDB

### Database version

10.6

### Summary

Should I put it in the same location as...?
`bin  config  data  LICENSE  module  public  README.md  vendor`

What is the exact filename?
Is it .env?

Thank you so much.

**Comment by acelaya** [comment, marked as answer]
Shlink does not support .env files

**Comment by droplinxuser** [comment]
What I mean is this:
https://shlink.io/documentation/environment-variables

**Reply by acelaya** [reply]
You need to provide environment variables by any other means. Shlink does not actively look for a .env file

**Reply by droplinxuser** [reply]
With RoadRunner, I assume something like:
--dotenv /var/www/short.co/.env

What if I'm serving Shlink with a classic web server?

**Reply by acelaya** [reply]
It depends on the system, the web server, if you want those env vars to be available when running Shlink from the CLI, etc.

A quick Google search on "how to set environment variables..." should throw some results.

**Comment by droplinxuser** [comment]
Is there any way I can alter the configuration options without running:
vendor/bin/shlink-installer install

Like directly editing a config file.
I'm not sure which file to modify.

**Reply by acelaya** [reply]
Shlink installer is literally all you need to set the configuration or edit it, if you don't want to use env vars.

Refrain from manually editing Shlink sources.

**Reply by droplinxuser** [reply]
Same for these 2 files; should I not edit them manually?
/var/www/xx/data/cache/app_config.php
/var/www/xx/config/params/generated_config.php

**Comment by droplinxuser** [comment]
From here:
https://shlink.io/documentation/install-dist-file/

Is says:
> Providing your configuration via [env vars](https://shlink.io/documentation/environment-variables) (since v2.9.0). Useful to pre-define the configuration or if you don’t have shell access to the server.

I'm not sure how to do that.

**Reply by acelaya** [reply]
Setting environment variables deeply depends on the system you use, but has nothing to do with Shlink per se.

I would recommend you to start by googling how environment variables work, and how to set them in your specific setup.
